src/core_plugins/elasticsearch/lib/cluster.js

```diff
@@ -0,0 +1,137 @@
+import elasticsearch from 'elasticsearch';
+import { get, set, isEmpty, cloneDeep, pick } from 'lodash';
+import toPath from 'lodash/internal/toPath';
+import Boom from 'boom';
+
+import filterHeaders from './filter_headers';
+import { parseConfig } from './parse_config';
+
+export default class Cluster {
+  constructor(config) {
+    this._config = Object.assign({}, config);
+    this.errors = elasticsearch.errors;
+
+    createClients.call(this);
+
+    return this;
+  }
+
+  /**
+   * callWithRequest
+   *
+   * Performs a call to ES, passing through whitelisted headers in the request
+   *
+   * @param {Object|undefined} req - The request object
+   * @param {string} endpoint
+   * @param {Object} clientParams
+   * @param {Object} options
+   * @param {boolean} options.wrap401Errors
+   * @returns {Promise}
+   */
+  callWithRequest = (req = {}, endpoint, clientParams = {}, options = {}) => {
```

Comment:
Ah, I understand what you're saying, but I'm honestly surprised this is working at all, and I worry it only works because of babel.  Arrow functions adopt the scope in which they are defined, which in this case appears to be in the scope of the module itself rather than a specific instance of that class.

Or perhaps this is actually the intended behavior for the class property syntax?  I'm not really familiar with it since it's still in proposal stage.